src/status.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""The charm status module."""
+
+import typing
+
+import ops
+
+
+def get_priority_status(statuses: typing.Iterable[ops.StatusBase]) -> ops.StatusBase:
+    """Get status to display out of all possible statuses returned by charm components.
+
+    Args:
+        statuses: Statuses returned by components of the charm.
+
+    Returns:
+        The final status to display.
+    """
+
+    def get_status_priority(status: ops.StatusBase) -> int:
```

Comment:
What is the reason for this function to be embedded in the `get_priority_status` function?

Reply:
This function has been refactored as per the suggestion, thank you!